### PR TITLE
New package: fjcontrib + new variants for fastjet

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/fjcontrib/package.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    def build(self, spec, prefix):
+        with working_dir(self.build_directory):
+            for target in self.build_targets:
+                inspect.getmodule(self).make(target)
+
+    def install(self, spec, prefix):
+        with working_dir(self.build_directory):
+            for target in self.install_targets:
+                inspect.getmodule(self).make(target)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/fjcontrib/package.py:85]: This seems to be the default for AutotoolsPackage, although the base class will run `make('install', 'fragile-shared-install')`, not sure if it needs to be run sequentially.

File: var/spack/repos/builtin/packages/fjcontrib/package.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    depends_on('fastjet')
+
+    build_targets = ['all', 'fragile-shared']
+    install_targets = ['install', 'fragile-shared-install']
+
+    def configure_args(self):
+        args = ['--fastjet-config=' +
+                self.spec['fastjet'].prefix.bin +
+                '/fastjet-config',
+                "CXXFLAGS=-O3 -Wall -g -fPIC"]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/fjcontrib/package.py]: Use `self.compiler.cxx_pic_flag` so this works for every compiler

**Review by adamjstewart — COMMENTED** [review]
@drbenmorgan @vvolkl can you review the changes to fastjet?

File: var/spack/repos/builtin/packages/fjcontrib/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/fjcontrib/package.py:85]: Yes, it has to be ran sequentially.

**Review by vvolkl — APPROVED** [review]
Thanks @iarspider  - looks good to me.

File: var/spack/repos/builtin/packages/fjcontrib/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/fjcontrib/package.py:85]: Hmm, I wonder if we should change this in the `AutotoolsPackage` base class then... @alalazo any thoughts?

**Review by adamjstewart — APPROVED** [review]

**Comment by drbenmorgan** [comment]
Looks o.k. to me, but could you confirm what the additional variants are for @iarspider please? The shared one looks ok., but not clear about the other two. For example, would `allplugins` be better as a default on no-variant option?

**Comment by iarspider** [comment]
@drbenmorgan good catch! 

@adamjstewart I hope I've used `enable_or_disable` correctly, could you please check?

**Comment by drbenmorgan** [comment]
Thanks @iarspider, but I wondered if it's o.k. to enable all the plugins by default and remove the variant? That would make it easier for the user, unless there's a big cost in compile time/disk space to enable all of them?

**Comment by iarspider** [comment]
As far as I can tell, the disk usage and build time are very small. OK, I will update the recipe.

**Comment by drbenmorgan** [comment]
Thanks @iarspider, looks good now, just the small Flake8 error to fix, then it's good to merge!

**Pull request merged**